### PR TITLE
Porting DTNNModel Class

### PR DESCRIPTION
**Pull request by GreatRSingh**
## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by shaipranesh2 — APPROVED** [review]

**Review by tonydavis629 — APPROVED** [review]
This LGTM with a rebase

File: deepchem/models/tests/test_dtnn.py [hunk 1]
```diff
@@ -47,3 +49,33 @@ def test_dtnn():
 
     # Check number of parameters
     assert len(list(model.parameters())) == 17
+
+
+@pytest.mark.torch
```
Comment (rbharath) [inline deepchem/models/tests/test_dtnn.py:54]: We need save/reload/overfit tests. See https://deepchem.readthedocs.io/en/latest/development_guide/coding.html#testing-machine-learning-models

File: deepchem/models/torch_models/dtnn.py [hunk 2]
```diff
@@ -145,3 +148,134 @@ def forward(self, inputs: List[torch.Tensor]):
         dtnn_gather = nn.Dropout(self.dropout)(dtnn_gather)
         output = self.linear(dtnn_gather)
         return output
+
+
+class DTNNModel(TorchModel):
+    """Implements DTNN models for regression.
+
+    This class implements the Deep Tensor Neural Network (DTNN) [1]_.
```
Comment (rbharath) [inline deepchem/models/torch_models/dtnn.py:156]: Can you add 1-2 paragraphs of explanation? Explain the core idea of the model and how it works

File: docs/source/api_reference/models.rst [hunk 3]
```diff
@@ -484,6 +484,11 @@ GroverModel
 .. autoclass:: deepchem.models.torch_models.GroverModel
   :members:
 
+DTNNModel
```
Comment (rbharath) [inline docs/source/api_reference/models.rst:487]: You don't need a new section for DTNNModel in the docs

File: deepchem/models/tests/test_overfit.py [hunk 4]
```diff
@@ -1026,3 +1026,42 @@ def test_textCNN_singletask_regression_overfit():
...
+    scores = model.evaluate(dataset, [regression_metric])
+    assert scores[regression_metric.name] > .7
+
+
+@flaky()
+@pytest.mark.torch
+def test_dtnn_multitask_regression_overfit():
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    dataset_file = os.path.join(current_dir, "assets/qm9_mini.sdf")
+    TASKS = ["alpha"]
```
Comment (rbharath) [inline deepchem/models/tests/test_overfit.py:1055]: You may need to add to another task here

**Review by rbharath — COMMENTED** [review]
Couple of minor issues (test fixes, more docs)

File: deepchem/models/torch_models/dtnn.py (hunk 2 above)
Comment (GreatRSingh) [reply deepchem/models/torch_models/dtnn.py:161]: Done

File: deepchem/models/tests/test_overfit.py (hunk 4 above)
Comment (GreatRSingh) [reply deepchem/models/tests/test_overfit.py]: Done, Sorry about that.

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**